src/serializer/ResidualHeapSerializer.js

```diff
@@ -1650,10 +1650,10 @@ export class ResidualHeapSerializer {
       invariant(abstractIndex >= 0 && abstractIndex < val.args.length);
       return serializedArgs[abstractIndex];
     }
-    if (val.kind === "implicit conversion to object") {
+    if (val.kind === "explicit conversion to object") {
       let ob = serializedArgs[0];
       invariant(ob !== undefined);
-      return ob;
+      return t.callExpression(this.preludeGenerator.memoizeReference("Object"), [ob]);
```

Comment:
IMO, it's a bit better to reference the intrinsic value since we don't know if this context is going to be shadowed or even if the global property has later been renamed.

```js
function foo(Object) {
  return Object.x;
}
```

Change to `realm.intrinsics. Object`.

Reply:
This call is not equivalent to `ToObject` because `ToObject` throws for `null`/`undefined`.

Where as this would return an empty object:

https://github.com/facebook/prepack/blob/master/src/intrinsics/ecma262/Object.js#L53

You'd have to check for null/undefined.

Reply:
Good catch. Looks like Object.assign will do the right thing.

Reply:
memoizeReference already takes care of the shadowing problem.